### PR TITLE
[Workflow]: Auto build Official Docker Image

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

Auto build official docker image by github workflow actions. 

## Brief change log

- Add workflow to build official image Automated.

1. When commits merged into master or tags pushed,  `amoro`, `optimizer-flink`, `quickdemo` will be built automated.
2. When a PR is merged into the master branch, it triggers the building of a "master-snapshot" image. When a tag is pushed, it triggers the building of a "release" image, and the image tag matches the corresponding GitHub tag.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by shidayang — APPROVED** [review]

**Comment by shidayang** [comment]
LGTM

File: .github/workflows/docker-images.yml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+on:
+  push:
+    branches:
+      - "master"
+    tags:
+      - "v*"
+
+
+  workflow_dispatch:
```
Comment (zhoujinsong) [inline .github/workflows/docker-images.yml:32]: What is the `on:workflow_dispatch` used for?

Comment (baiyangtx) [reply .github/workflows/docker-images.yml]: fixed.

File: .github/workflows/docker-images.yml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+# limitations under the License.
+#
+
+
+# This workflow will build docker images when commit merged or pushed to master.
+# or tags pushed.
+
+name: Publish Docker Image
+
+on:
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply .github/workflows/docker-images.yml:24]: fixed.

File: .github/workflows/docker-images.yml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+          context: .
+          push: true
+          file: docker/optimizer-flink/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          tags: ${{ steps.meta.outputs.tags }}
+          build-args: |
+            FLINK_VERSION=${{ matrix.flink }}
+            OPTIMIZER_JOB=ams/optimizer/flink-optimizer/target/flink-optimizer-${{ env.AMORO_VERSION }}-jar-with-dependencies.jar
+
+  docker-quickdemo:
```
Comment (zhoujinsong) [inline .github/workflows/docker-images.yml:175]: Maybe we should build the quick demo image only for new tags.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**